### ISSUE TITLE
static and class methods in a custom exception lose their first argument

### ISSUE THREAD

**Issue by scottbelden** [agronholm/sphinx-autodoc-typehints#23]
If I have something like this:

```python
class MyException(Exception):
    @classmethod
    def from_name_and_number(cls,
                             name: str,
                             number: int):
        """
        Args:
            name: name
            number: number
        """
        return cls(f'{name}: {number}')

    @staticmethod
    def is_same(name: str,
                number: int) -> bool:
        """
        Args:
            name: name
            number: number
        """
        return str(number) == name
```

When I generate the documentation I get the following:
```
classmethod from_name_and_number(number)
    Parameters: * name(str) - name
                * number(int) - number

static is_same(number)
    Parameters: * name(str) - name
                * number(int) - number

    Return type: bool
```

Notice that both the functions look like they only take in a `number` argument. I think this is specific to `Excpetions` as I don't think I see the same issues for other methods.

For reference, if I run without this extension I get something like the following:
```
classmethod from_name_and_number(name: str, number: int)
    Parameters: * name - name
                * number - number

static is_same(name: str, number: int) → bool
    Parameters: * name - name
                * number - number
```

**Comment by scottbelden** [comment]
I'm pretty sure it was the case, but I'd have to set up the example I posted above again to know for sure. As it turns out I realized that I didn't really need the static or class methods anyways, so this isn't really important to me anymore, but I didn't close it as I believe it is an issue.

**Comment by agronholm** [comment]
Oddly enough, the problem seems to be easily fixable for class methods, as they appear as bound methods. But I have not yet found a way to distinguish between static and instance methods.

**Comment by agronholm** [comment]
Fixed via 4fd7cff2abe446a10567f75fab2b290a381d1631.
